packages/storage/test/unit/reference.exp.test.ts

```diff
@@ -0,0 +1,556 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { assert } from 'chai';
+import { FirebaseApp } from '@firebase/app-types';
+import { StringFormat } from '../../src/implementation/string';
+import { Headers } from '../../src/implementation/xhrio';
+import { Metadata } from '../../src/metadata';
+import {
+  Reference,
+  uploadString,
+  uploadBytes,
+  deleteObject,
+  listAll,
+  list,
+  getMetadata,
+  updateMetadata,
+  getParent as parentReference,
+  getDownloadURL
+} from '../../src/reference';
+import { StorageService, ref } from '../../src/service';
+import * as testShared from './testshared';
+import { SendHook, TestingXhrIo } from './xhrio';
+import { DEFAULT_HOST } from '../../src/implementation/constants';
+import { FirebaseAuthInternalName } from '@firebase/auth-interop-types';
+import { Provider } from '@firebase/component';
+
+/* eslint-disable @typescript-eslint/no-floating-promises */
+function makeFakeService(
+  app: FirebaseApp,
+  authProvider: Provider<FirebaseAuthInternalName>,
+  sendHook: SendHook
+): StorageService {
+  return new StorageService(app, authProvider, testShared.makePool(sendHook));
+}
+
+function makeStorage(url: string): Reference {
+  const service = new StorageService(
+    {} as FirebaseApp,
+    testShared.emptyAuthProvider,
+    testShared.makePool(null)
+  );
+  return new Reference(service, url);
+}
+
+describe('Firebase Storage > Reference', () => {
+  const root = makeStorage('gs://test-bucket/');
+  const child = makeStorage('gs://test-bucket/hello');
+  describe('Path constructor', () => {
+    it('root', () => {
+      assert.equal(root.toString(), 'gs://test-bucket/');
```

Comment:
We should use `expect` instead of `assert` in new tests to be consistent with other SDKs. We should eventually convert old tests to use `expect` as well.

Reply:
Wrote a script to replace these, but only ran it on the 4 "new" files to avoid making this PR too big.